rskj-core/src/main/java/co/rsk/db/RepositoryLocator.java

```diff
@@ -45,6 +59,11 @@ public Repository startTrackingAt(BlockHeader header) {
 
     private MutableTrie mutableTrieSnapshotAt(BlockHeader header) {
         Keccak256 stateRoot = stateRootHandler.translate(header);
-        return new MutableTrieImpl(repository.getTrie().getSnapshotTo(stateRoot));
+
+        if (EMPTY_HASH.equals(stateRoot)) {
```

Comment:
I cannot find this behavior in the `co.rsk.db.StateRootHandler#translate` or in the context of this commit, how did you get this?

Reply:
From here:

```java
    public Trie getSnapshotTo(Keccak256 hash) {
        if (emptyHash.equals(hash)) {
            return new Trie(this.store);
        }
```